### PR TITLE
JSC for Android

### PR DESCRIPTION
**Pull request by piaskowyk**
## Description

Added support for JSC runtime on Android

## Checklist

- [x] Ensured that CI passes

### CODE REVIEW

File: Common/cpp/NativeModules/NativeReanimatedModule.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ std::vector<std::shared_ptr<MutableValue>> extractMutablesFromArray(jsi::Runtime
 
 NativeReanimatedModule::NativeReanimatedModule(std::shared_ptr<CallInvoker> jsInvoker,
                                                std::shared_ptr<Scheduler> scheduler,
-                                               std::unique_ptr<jsi::Runtime> rt,
+                                               std::unique_ptr<jsi::Runtime>& rt,
```
Comment (kmagiera) [inline Common/cpp/NativeModules/NativeReanimatedModule.cpp:61]: this does not seem like a good idea, if the point of the pointer is to be referenced in a number of places we should use shared_ptr and not unique_ptr -> unique is for usecases where there is only one "holder"

**Review by Szymon20000 — APPROVED** [review]
Looks fine to me.

**Comment by Yonom** [comment]
@piaskowyk do you have an ETA on when this will land?

**Comment by piaskowyk** [comment]
@Yonom We released it in the latest version [2.1.0](https://github.com/software-mansion/react-native-reanimated/releases/tag/2.1.0).

**Comment by zachgibson** [comment]
Does this PR remove the need to use Hermes on Android?

**Comment by piaskowyk** [comment]
@zachgibson Yes, you can choose JSC or Hermes runtime for Android

**Comment by SaeedZhiany** [comment]
@piaskowyk 

What about iOS, can also we use this library without Hermes enabled for iOS? if yes what is the minimum version that can be used?

**Comment by Arkan4ik** [comment]
@SaeedZhiany I am using "react-native-reanimated": "^2.3.1", and "react-native": "0.64.2", without hermes enabled and all works fine

**Comment by HSReact** [comment]
> @SaeedZhiany I am using "react-native-reanimated": "^2.3.1", and "react-native": "0.64.2", without hermes enabled and all work fine

Hi @Arkan4ik how is the performance of reanimated 2 without hermes? is it better than reanimated 1? Thanks for any help

**Pull request merged**